Commit ID: f0cf3176a2d7d61f89f52844c38e0081b1492e0f
Change ID: wlvsqkmvonzutxuxrsuolqxyrlvwowtk
Author   : Jonas Fierlings <fnoegip@gmail.com> (2026-01-11 20:27:33)
Committer: Jonas Fierlings <fnoegip@gmail.com> (2026-07-19 14:26:04)
Signature: (no signature)

    ServicesCapAdd patch

    Signed-off-by: Jonas Fierlings <fnoegip@gmail.com>

diff --git a/common/config.go b/common/config.go
index 575ef6b505..c2c345a273 100644
--- a/common/config.go
+++ b/common/config.go
@@ -311,6 +311,7 @@
 	DeviceCgroupRules          []string            `toml:"device_cgroup_rules,omitempty" json:"device_cgroup_rules,omitempty" long:"device-cgroup-rules" env:"DOCKER_DEVICE_CGROUP_RULES" description:"Add a device cgroup rule to the container"`
 	Gpus                       string              `toml:"gpus,omitempty" json:"gpus" long:"gpus" env:"DOCKER_GPUS" description:"Request GPUs to be used by Docker"`
 	ServicesDevices            map[string][]string `toml:"services_devices,omitempty" json:"services_devices,omitempty" long:"services_devices" env:"DOCKER_SERVICES_DEVICES" description:"A toml table/json object with the format key=values. Expose host devices to services based on image name."`
+	ServicesCapAdd             []string            `toml:"services_cap_add" json:"services_cap_add,omitempty" long:"services-cap-add" env:"DOCKER_SERVICES_CAP_ADD" description:"Add Linux capabilities to services"`
 	DisableCache               bool                `toml:"disable_cache,omitzero" json:"disable_cache" long:"disable-cache" env:"DOCKER_DISABLE_CACHE" description:"Disable all container caching"`
 	Volumes                    []string            `toml:"volumes,omitempty" json:"volumes,omitempty" long:"volumes" env:"DOCKER_VOLUMES" description:"Bind-mount a volume and create it if it doesn't exist prior to mounting. Can be specified multiple times once per mountpoint, e.g. --docker-volumes 'test0:/test0' --docker-volumes 'test1:/test1'"`
 	VolumeKeep                 bool                `toml:"volume_keep,omitzero" json:"volume_keep" long:"volume-keep" env:"DOCKER_VOLUME_KEEP" description:"Do not delete volumes on container removal. Enabling can lead to increase in storage"`
diff --git a/executors/docker/docker.go b/executors/docker/docker.go
index 3b727eb89f..41ccfecf45 100644
--- a/executors/docker/docker.go
+++ b/executors/docker/docker.go
@@ -668,6 +668,7 @@
 		ExtraHosts:    e.Config.Docker.ExtraHosts,
 		Privileged:    privileged,
 		SecurityOpt:   servicesSecurityOpt,
+		CapAdd:        e.Config.Docker.ServicesCapAdd,
 		Runtime:       e.Config.Docker.Runtime,
 		UsernsMode:    container.UsernsMode(e.Config.Docker.UsernsMode),
 		NetworkMode:   e.networkMode,
